assets/js/plugin-check-admin.js

```diff
@@ -172,14 +172,41 @@
 	 * @param {Object} data The response data.
 	 */
 	async function runChecks( data ) {
+		let isSuccessMessage = true;
+		let messageClass = '';
+		let messageText = '';
 		for ( let i = 0; i < data.checks.length; i++ ) {
 			try {
 				const results = await runCheck( data.plugin, data.checks[ i ] );
+				const errorsLength = Object.values( results.errors ).length;
+				const warningsLength = Object.values( results.warnings ).length;
+				if (
+					isSuccessMessage &&
+					( errorsLength > 0 || warningsLength > 0 )
+				) {
+					isSuccessMessage = false;
+				}
 				renderResults( results );
 			} catch ( e ) {
 				// Ignore for now.
 			}
 		}
+
+		if ( isSuccessMessage ) {
+			messageClass = 'notice-success';
+			messageText = pluginCheck.successMessage;
+		} else {
+			messageClass = 'notice-error';
+			messageText = pluginCheck.errorMessage;
+		}
```

Comment:
@mukeshpanchal27 see my point above, if we default to a success message we can simply check for if success is false and update accordingly.

```suggestion
		if ( ! isSuccessMessage ) {
			messageClass = 'notice-error';
			messageText = pluginCheck.errorMessage;
		}
```